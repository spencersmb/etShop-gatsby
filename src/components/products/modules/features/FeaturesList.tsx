import FeatureItem from '@components/products/modules/features/featureItem'
import { IFeatureItem } from '@et/types/Products'
import { device } from '@styles/global/breakpoints'
import { colors } from '@styles/global/colors'
import { GridFluid } from '@styles/global/cssGrid'
import { Sentinel } from '@styles/global/fonts'
import { svgs } from '@svg'
import { renderSvg } from '@utils/styleUtils'
import React from 'react'
import styled from 'styled-components'

interface IProps {
	features: IFeatureItem[]
}

const FeaturesList = (props: IProps) => {

	const { features } = props
	return (
		<Container>
			<TitleHeader data-testid={'title'}>
				Features
			</TitleHeader>
			<Features data-testid={'items'}>
				{features.map((feature, index) => (<FeatureItem index={index} key={index} {...feature}/>))}
			</Features>
			<Divider>
				{renderSvg(svgs.Divider1)}
			</Divider>
		</Container>
	)
}
const Features = styled.div`
	grid-column: 2 /4;
	position: relative;
	z-index: 2;
	align-items: flex-start;
	@media ${device.tablet} {
		grid-column: 3 / 13;
	}
	
	@media ${device.laptop} {
		grid-column: 2 / 14;
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-template-rows: 80px 1fr 80px 1fr;
		//flex-direction: row;
		//flex-wrap: wrap;
		//align-items: flex-start;
	}
		
`
const Divider = styled.div`
width: 100%;
position: absolute;
bottom:0;
left:0;
z-index: 1;
height: 350px;
overflow: hidden;
grid-column: 1 / -1;
	svg{
		width: 850px;
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
	}
	
	@media ${device.tablet} {
		svg{
			width: 1500px;
		}
	}
	
	@media ${device.laptop} {
		svg{
			width: 2020px;
		}
	}
	
		@media ${device.laptopL} {
		svg{
			width: 3020px;
		}
	}
	
`
const Container = styled(GridFluid)`
grid-row-gap: 0 !important;
padding: 60px 0 80px;
overflow: hidden;
position: relative;
z-index: 1;

	@media ${device.laptop} {
		margin-top: 80px;
		padding: 110px 0 150px;
	}
	@media ${device.laptopL} {
		padding: 190px 0 180px;
	}
`
const TitleHeader = styled.h5`
	font-weight: 500;
	${Sentinel.semiboldItalic};
	font-style: italic;
	position: relative;
	color: ${colors.primary.headline};
	grid-column: 2 /4;
	margin-bottom: 25px;
	font-size: 28px;
	z-index: 2;
	text-align: center;
	
	@media ${device.tablet} {
    font-size: 34px;
    font-weight: 500;
    font-style: italic;
    margin-bottom: 25px;
    text-align: center;
    grid-column: 2 / 14;
	}
	
	@media ${device.laptop} {
		color: #d6e0ea;
		left: 0;
		top: -30px;
		font-size: 190px;
    line-height: 190px;
    position: absolute;
	}
	@media ${device.laptopL} {
		top: 0;
		font-size: 248px;
    line-height: 248px;
	}
		
`
export default React.memo(FeaturesList)
