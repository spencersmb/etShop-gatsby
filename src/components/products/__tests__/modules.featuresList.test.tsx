import FeaturesList from '@components/products/modules/features/FeaturesList'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	features: [
		{
			description: 'description',
			icon: 'texture',
			title: 'title 1'
		},
		{
			description: 'description',
			icon: 'grid',
			title: 'title 2'
		}
	]
}
describe('Feature List Tests', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<FeaturesList {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct title', () => {
		const modalRender = render(<FeaturesList {...propsDefault}/>)
		expect(modalRender.getByTestId('title').innerHTML).toEqual('Features')
	})
	it('Should render correct # of items', () => {
		const modalRender = render(<FeaturesList {...propsDefault}/>)
		expect(modalRender.getByTestId('items').children.length).toEqual(propsDefault.features.length)
	})
})
