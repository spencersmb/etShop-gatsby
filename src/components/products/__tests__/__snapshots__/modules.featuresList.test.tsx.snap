// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature List Tests renders correctly 1`] = `
<div
  className="sc-ifAKCX sc-dnqmqq kLOfep"
>
  <h5
    className="sc-iwsKbI eHxWpl"
    data-testid="title"
  >
    Features
  </h5>
  <div
    className="sc-gzVnrw QqsNn"
    data-testid="items"
  >
    <div
      className="sc-bxivhb fyrUIa"
    >
      <div
        className="sc-htpNat ePPfSV"
        color="#A182E9"
        data-testid="title"
      >
        title 1
      </div>
      <div
        className="sc-bwzfXH ihTQNy"
        data-testid="desc"
      >
        description
      </div>
      <div
        className="sc-bdVaJa ktqiA"
        color="#A182E9"
      >
        <test-file-stub />
      </div>
    </div>
    <div
      className="sc-bxivhb fyrUIa"
    >
      <div
        className="sc-htpNat gYWYhm"
        color="#18C3C3"
        data-testid="title"
      >
        title 2
      </div>
      <div
        className="sc-bwzfXH ihTQNy"
        data-testid="desc"
      >
        description
      </div>
      <div
        className="sc-bdVaJa hkMwfE"
        color="#18C3C3"
      >
        <test-file-stub />
      </div>
    </div>
  </div>
  <div
    className="sc-htoDjs ihmdgm"
  >
    <test-file-stub />
  </div>
</div>
`;
